--- a/shell/ash.c
+++ b/shell/ash.c
@@ -16282,7 +16282,14 @@ int ash_main(int argc UNUSED_PARAM, char **argv)
 
 		state = 1;
 #if ENABLE_PLATFORM_MINGW32
-		hp = xasprintf("%s/etc/profile", get_system_drive() ?: "");
+		const char *syspath = bb_busybox_exec_path;
+		int syspathlen = 0;
+		for (int i = 0; syspath[i]; i++) {
+			if (syspath[i] == '/') {
+				syspathlen = i;
+			}
+		}
+		hp = xasprintf("%.*s/../src/profile", syspathlen, syspath);
 		read_profile(hp);
 		free((void *)hp);
 #else
