--- a/Dockerfile
+++ b/Dockerfile
@@ -62,3 +62,3 @@ COPY src/w64devkit.c src/w64devkit.ico \
 
-ARG ARCH=x86_64-w64-mingw32
+ARG ARCH=i686-w64-mingw32
 
@@ -87,2 +87,3 @@ RUN /mingw-w64-v$MINGW_VERSION/mingw-w64-headers/configure \
         --host=$ARCH \
+        --with-default-win32-winnt=0x0501 \
  && make -j$(nproc) \
@@ -100,2 +101,3 @@ RUN cat $PREFIX/src/gcc-*.patch | patch -d/gcc-$GCC_VERSION -p1 \
         --target=$ARCH \
+        --with-arch=pentium4 \
         --enable-static \
@@ -127,4 +129,4 @@ RUN /mingw-w64-v$MINGW_VERSION/mingw-w64-crt/configure \
         --disable-dependency-tracking \
-        --disable-lib32 \
-        --enable-lib64 \
+        --enable-lib32 \
+        --disable-lib64 \
         CFLAGS="-Os" \
@@ -211,2 +213,3 @@ RUN /mingw-w64-v$MINGW_VERSION/mingw-w64-headers/configure \
         --host=$ARCH \
+        --with-default-win32-winnt=0x0501 \
  && make -j$(nproc) \
@@ -220,4 +223,4 @@ RUN /mingw-w64-v$MINGW_VERSION/mingw-w64-crt/configure \
         --disable-dependency-tracking \
-        --disable-lib32 \
-        --enable-lib64 \
+        --enable-lib32 \
+        --disable-lib64 \
         CFLAGS="-Os" \
@@ -246,2 +249,3 @@ RUN /gcc-$GCC_VERSION/configure \
         --host=$ARCH \
+        --with-arch=pentium4 \
         --enable-static \
