--- a/Dockerfile
+++ b/Dockerfile
@@ -74,3 +74,3 @@ COPY src/w64devkit.c src/w64devkit.ico \
 
-ARG ARCH=x86_64-w64-mingw32
+ARG ARCH=i686-w64-mingw32
 
@@ -102,2 +102,3 @@ RUN /mingw-w64-v$MINGW_VERSION/mingw-w64-headers/configure \
         --with-default-msvcrt=msvcrt-os \
+        --with-default-win32-winnt=0x0501 \
  && make -j$(nproc) \
@@ -108,2 +109,5 @@ RUN ln -s $ARCH mingw
 
+# Disable UTF-8 manifest for Windows XP (#58)
+RUN echo >/gcc-$GCC_VERSION/gcc/config/i386/winnt-utf8.manifest
+
 WORKDIR /x-gcc
@@ -113,2 +117,3 @@ RUN /gcc-$GCC_VERSION/configure \
         --target=$ARCH \
+        --with-arch=pentium4 \
         --enable-static \
@@ -141,4 +146,4 @@ RUN /mingw-w64-v$MINGW_VERSION/mingw-w64-crt/configure \
         --disable-dependency-tracking \
-        --disable-lib32 \
-        --enable-lib64 \
+        --enable-lib32 \
+        --disable-lib64 \
         CFLAGS="-Os" \
@@ -226,2 +231,3 @@ RUN /mingw-w64-v$MINGW_VERSION/mingw-w64-headers/configure \
         --with-default-msvcrt=msvcrt-os \
+        --with-default-win32-winnt=0x0501 \
  && make -j$(nproc) \
@@ -236,4 +242,4 @@ RUN /mingw-w64-v$MINGW_VERSION/mingw-w64-crt/configure \
         --disable-dependency-tracking \
-        --disable-lib32 \
-        --enable-lib64 \
+        --enable-lib32 \
+        --disable-lib64 \
         CFLAGS="-Os" \
@@ -248,2 +254,3 @@ RUN /mingw-w64-v$MINGW_VERSION/mingw-w64-libraries/winpthreads/configure \
         --host=$ARCH \
+        --with-arch=pentium4 \
         --enable-static \
@@ -512,3 +519,3 @@ RUN cat $PREFIX/src/git-*.patch | patch -p1 \
        NO_GITWEB=1 \
-       MSYSTEM=MINGW64 \
+       MSYSTEM=MINGW32 \
        CFLAGS="-Os -I/deps/include/" \
