--- a/Dockerfile
+++ b/Dockerfile
@@ -102,3 +102,3 @@ RUN cat $PREFIX/src/gcc-*.patch | patch -d/gcc-$GCC_VERSION -p1 \
         --with-pic \
-        --enable-languages=c,c++ \
+        --enable-languages=c,c++,fortran \
         --enable-libgomp \
@@ -254,3 +254,3 @@ RUN sed -i 's#=/mingw/include#=/include#' /gcc-$GCC_VERSION/gcc/config.gcc \
         --with-mpfr-lib=/deps/lib \
-        --enable-languages=c,c++ \
+        --enable-languages=c,c++,fortran \
         --enable-libgomp \
@@ -288,3 +288,3 @@ RUN $ARCH-gcc -DEXE=gcc.exe -DCMD=cc \
       gcc gcc-ar gcc-nm gcc-ranlib gcov gcov-dump gcov-tool ld nm objcopy \
-      objdump ranlib readelf size strings strip windmc windres \
+      objdump ranlib readelf size strings strip windmc windres gfortran \
     | xargs -I{} -P$(nproc) \
