--- a/Dockerfile
+++ b/Dockerfile
@@ -53,7 +53,7 @@ RUN sha256sum -c $PREFIX/src/SHA256SUMS \
 COPY src/w64devkit.c src/w64devkit.ico src/alias.c src/debugbreak.c \
      $PREFIX/src/
 
-ARG ARCH=x86_64-w64-mingw32
+ARG ARCH=i686-w64-mingw32
 
 # Build cross-compiler
 
@@ -78,6 +78,7 @@ WORKDIR /x-mingw-headers
 RUN /mingw-w64-v$MINGW_VERSION/mingw-w64-headers/configure \
         --prefix=/bootstrap/$ARCH \
         --host=$ARCH \
+        --with-default-win32-winnt=0x0501 \
  && make -j$(nproc) \
  && make install
 
@@ -89,6 +90,7 @@ RUN /gcc-$GCC_VERSION/configure \
         --prefix=/bootstrap \
         --with-sysroot=/bootstrap \
         --target=$ARCH \
+        --with-arch=pentium4 \
         --enable-static \
         --disable-shared \
         --with-pic \
@@ -116,8 +118,8 @@ RUN /mingw-w64-v$MINGW_VERSION/mingw-w64-crt/configure \
         --with-sysroot=/bootstrap/$ARCH \
         --host=$ARCH \
         --disable-dependency-tracking \
-        --disable-lib32 \
-        --enable-lib64 \
+        --enable-lib32 \
+        --disable-lib64 \
         CFLAGS="-Os" \
         LDFLAGS="-s" \
  && make -j$(nproc) \
@@ -199,6 +201,7 @@ WORKDIR /mingw-headers
 RUN /mingw-w64-v$MINGW_VERSION/mingw-w64-headers/configure \
         --prefix=$PREFIX/$ARCH \
         --host=$ARCH \
+        --with-default-win32-winnt=0x0501 \
  && make -j$(nproc) \
  && make install
 
@@ -208,8 +211,8 @@ RUN /mingw-w64-v$MINGW_VERSION/mingw-w64-crt/configure \
         --with-sysroot=$PREFIX/$ARCH \
         --host=$ARCH \
         --disable-dependency-tracking \
-        --disable-lib32 \
-        --enable-lib64 \
+        --enable-lib32 \
+        --disable-lib64 \
         CFLAGS="-Os" \
         LDFLAGS="-s" \
  && make -j$(nproc) \
@@ -234,6 +237,7 @@ RUN sed -i 's#=/mingw/include#=/include#' /gcc-$GCC_VERSION/gcc/config.gcc \
         --with-sysroot=$PREFIX/$ARCH \
         --target=$ARCH \
         --host=$ARCH \
+        --with-arch=pentium4 \
         --enable-static \
         --disable-shared \
         --with-pic \
