--- a/Dockerfile
+++ b/Dockerfile
@@ -61,3 +61,3 @@ COPY src/w64devkit.c src/w64devkit.ico src/alias.c src/debugbreak.c \
 
-ARG ARCH=x86_64-w64-mingw32
+ARG ARCH=i686-w64-mingw32
 
@@ -86,2 +86,3 @@ RUN /mingw-w64-v$MINGW_VERSION/mingw-w64-headers/configure \
         --host=$ARCH \
+        --with-default-win32-winnt=0x0501 \
  && make -j$(nproc) \
@@ -99,2 +100,3 @@ RUN cat $PREFIX/src/gcc-*.patch | patch -d/gcc-$GCC_VERSION -p1 \
         --target=$ARCH \
+        --with-arch=pentium4 \
         --enable-static \
@@ -126,4 +128,4 @@ RUN /mingw-w64-v$MINGW_VERSION/mingw-w64-crt/configure \
         --disable-dependency-tracking \
-        --disable-lib32 \
-        --enable-lib64 \
+        --enable-lib32 \
+        --disable-lib64 \
         CFLAGS="-Os" \
@@ -210,2 +212,3 @@ RUN /mingw-w64-v$MINGW_VERSION/mingw-w64-headers/configure \
         --host=$ARCH \
+        --with-default-win32-winnt=0x0501 \
  && make -j$(nproc) \
@@ -219,4 +222,4 @@ RUN /mingw-w64-v$MINGW_VERSION/mingw-w64-crt/configure \
         --disable-dependency-tracking \
-        --disable-lib32 \
-        --enable-lib64 \
+        --enable-lib32 \
+        --disable-lib64 \
         CFLAGS="-Os" \
@@ -231,2 +234,3 @@ RUN /mingw-w64-v$MINGW_VERSION/mingw-w64-libraries/winpthreads/configure \
         --host=$ARCH \
+        --with-arch=pentium4 \
         --enable-static \
