--- a/Dockerfile
+++ b/Dockerfile
@@ -2,7 +2,7 @@ FROM debian:bullseye-slim
 
 ARG VERSION=1.9.0 \
     PREFIX=/w64devkit \
-    ARCH=x86_64-w64-mingw32 \
+    ARCH=i686-w64-mingw32 \
     BINUTILS_VERSION=2.37 \
     BUSYBOX_VERSION=FRP-4264-gc79f13025 \
     CTAGS_VERSION=20200824 \
@@ -87,6 +87,7 @@ RUN /gcc-$GCC_VERSION/configure \
         --prefix=/bootstrap \
         --with-sysroot=/bootstrap \
         --target=$ARCH \
+        --with-arch=pentium4 \
         --enable-static \
         --disable-shared \
         --with-pic \
@@ -111,8 +112,8 @@ RUN /mingw-w64-v$MINGW_VERSION/mingw-w64-crt/configure \
         --with-sysroot=/bootstrap/$ARCH \
         --host=$ARCH \
         --disable-dependency-tracking \
-        --disable-lib32 \
-        --enable-lib64 \
+        --enable-lib32 \
+        --disable-lib64 \
         CFLAGS="-Os" \
         LDFLAGS="-s"
 RUN make -j$(nproc)
@@ -204,8 +205,8 @@ RUN /mingw-w64-v$MINGW_VERSION/mingw-w64-crt/configure \
         --with-sysroot=$PREFIX/$ARCH \
         --host=$ARCH \
         --disable-dependency-tracking \
-        --disable-lib32 \
-        --enable-lib64 \
+        --enable-lib32 \
+        --disable-lib64 \
         CFLAGS="-Os" \
         LDFLAGS="-s"
 RUN make -j$(nproc)
@@ -218,6 +219,7 @@ RUN /gcc-$GCC_VERSION/configure \
         --with-sysroot=$PREFIX \
         --target=$ARCH \
         --host=$ARCH \
+        --with-arch=pentium4 \
         --enable-static \
         --disable-shared \
         --with-pic \
