--- a/Dockerfile
+++ b/Dockerfile
@@ -90,7 +90,7 @@ RUN /gcc-$GCC_VERSION/configure \
         --enable-static \
         --disable-shared \
         --with-pic \
-        --enable-languages=c,c++ \
+        --enable-languages=c,c++,fortran \
         --enable-libgomp \
         --enable-threads=posix \
         --enable-version-specific-runtime-libs \
@@ -226,7 +226,7 @@ RUN sed -i 's#=/mingw/include#=/include#' /gcc-$GCC_VERSION/gcc/config.gcc \
         --with-mpc-lib=/deps/lib \
         --with-mpfr-include=/deps/include \
         --with-mpfr-lib=/deps/lib \
-        --enable-languages=c,c++ \
+        --enable-languages=c,c++,fortran \
         --enable-libgomp \
         --enable-threads=posix \
         --enable-version-specific-runtime-libs \
