--- a/Dockerfile
+++ b/Dockerfile
@@ -58,3 +58,3 @@ COPY src/w64devkit.c src/w64devkit.ico src/alias.c src/debugbreak.c \
 
-ARG ARCH=x86_64-w64-mingw32
+ARG ARCH=i686-w64-mingw32
 
@@ -83,2 +83,3 @@ RUN /mingw-w64-v$MINGW_VERSION/mingw-w64-headers/configure \
         --host=$ARCH \
+        --with-default-win32-winnt=0x0501 \
  && make -j$(nproc) \
@@ -96,2 +97,3 @@ RUN cat $PREFIX/src/gcc-*.patch | patch -d/gcc-$GCC_VERSION -p1 \
         --target=$ARCH \
+        --with-arch=pentium4 \
         --enable-static \
@@ -123,4 +125,4 @@ RUN /mingw-w64-v$MINGW_VERSION/mingw-w64-crt/configure \
         --disable-dependency-tracking \
-        --disable-lib32 \
-        --enable-lib64 \
+        --enable-lib32 \
+        --disable-lib64 \
         CFLAGS="-Os" \
@@ -206,2 +208,3 @@ RUN /mingw-w64-v$MINGW_VERSION/mingw-w64-headers/configure \
         --host=$ARCH \
+        --with-default-win32-winnt=0x0501 \
  && make -j$(nproc) \
@@ -215,4 +218,4 @@ RUN /mingw-w64-v$MINGW_VERSION/mingw-w64-crt/configure \
         --disable-dependency-tracking \
-        --disable-lib32 \
-        --enable-lib64 \
+        --enable-lib32 \
+        --disable-lib64 \
         CFLAGS="-Os" \
@@ -241,2 +244,3 @@ RUN sed -i 's#=/mingw/include#=/include#' /gcc-$GCC_VERSION/gcc/config.gcc \
         --host=$ARCH \
+        --with-arch=pentium4 \
         --enable-static \
